WebClient/Scripts/temperatureControlView.js

```diff
@@ -0,0 +1,503 @@
+﻿
+L.Control.Temp = L.Control.extend({
+    options: {
+        collapsed: false,
+        position: 'topleft',
+        autoZIndex: true,
+        height: 150,
+        width: 150,
+    },
+
+    /// leaflet constructor
+    initialize: function (options) {
+        L.setOptions(this, options);
+        this.parentContainer = options.element;
+    },
+
+    onAdd: function (map) {
+        this.active = true;
+        this._initLayout();
+        this._map = map;
+        return this._container;
+    },
+
+    onRemove: function () {
+        this.active = false;
+    },
+
+    _initLayout: function () {
```

Comment:
It would be great to split this function on parts like createMercuryBar, createTemperatureEdit, createUnitSelector where you will encapsulate all logic and variables that are neccessary to create those components. Ideally this function should just create one-few container divs and make 3-4 function calls to create your UI components.